libraries/AP_AdvancedFailsafe/AP_AdvancedFailsafe.cpp

```diff
@@ -253,7 +253,16 @@ AP_AdvancedFailsafe::check(uint32_t last_valid_rc_ms)
             _state = STATE_DATA_LINK_LOSS;
             if (_wp_comms_hold) {
                 _saved_wp = mission.get_current_nav_cmd().index;
-                mission.set_current_cmd(_wp_comms_hold);
+
+                //option to not go back to the loss comm mission item if we are already in return path or passed loss comm mission item
+                if ((mission.state() == AP_Mission::MISSION_RUNNING) &&
+                    ((option_is_set(Option::CONTINUE_IF_ALREADY_IN_RETURN_PATH) && mission.get_in_return_path_flag()) ||
+                     (option_is_set(Option::CONTINUE_IF_PASSED_LOSS_COMM_WP)    && (mission.get_current_nav_cmd().index > _wp_comms_hold)))){
```

Comment:
```suggestion
                if (should_use_comms_hold()) {
```

and

```
    bool AP_AdvancedFailsafe::should_use_comms_hold() const
    {
        if (mission.state() != AP_Mission::MISSION_RUNNING) {
            return false;
        }
        if (mission.get_in_return_path_flag() && option_is_set(Option::CONTINUE_IF_ALREADY_IN_RETURN_PATH))
            return false;
        }
        return true;
```

Please create a separate commit for adding this logic into the method using the same style:
```
                     (option_is_set(Option::CONTINUE_IF_PASSED_LOSS_COMM_WP)    && (mission.get_current_nav_cmd().index > _wp_comms_hold)))){
```

I'm not sure *that* patch will make it in easily, whereas I'm pretty confident of the return-path-start stuff.